src/main/java/com/fasterxml/jackson/core/util/TextBuffer.java

```diff
@@ -536,6 +536,37 @@ public char[] contentsAsArray() throws IOException {
      */
     public double contentsAsDouble(final boolean useFastParser) throws NumberFormatException {
         try {
+            if (_resultString == null) {
+                // Has array been requested? Can make a shortcut, if so:
+                if (_resultArray != null) {
+                    validateFPLength(_resultArray.length);
```

Comment:
Not needed: `StreamReadConstraints.validateFPLength()` is called from `ParserBase.resetFloat()` as part of decoding. `ReaderBasedJsonParser, line 1465 f.ex calls:

```
        _textBuffer.resetWithShared(_inputBuffer, startPtr, len);
        // And there we have it!
        return resetFloat(neg, intLen, fractLen, expLen);
```

so this is already taken care of